### ISSUE TITLE
Make suggestions based on Levenshtein distance

### ISSUE THREAD

**Issue by kinnison** [rust-lang/rustup#1816]
**Describe the problem you are trying to solve**

From time to time people have difficulty installing components because of typos.  Sometimes they're even really hard to spot, particularly for those who are dyslexic.  For example, recently someone had difficulty installing the `rls` component because they kept typing `rsl` but reading it as `rls` and so couldn't tell what was wrong.

**Describe the solution you'd like**

Whenever we issue a component-not-found or target-not-found etc, consider adding a quick check of the available components/targets, sort them by levenshtein distance, and suggest the nearest one.

**Notes**

That way, instead of:

```console
$ rustup component add rsl
error: toolchain 'stable-x86_64-unknown-linux-gnu' does not contain component 'rsl' for target 'x86_64-unknown-linux-gnu'
```

We'd maybe get something like:
```console
$ rustup component add rsl
error: toolchain 'stable-x86_64-unknown-linux-gnu' does not contain component 'rsl' for target 'x86_64-unknown-linux-gnu' - Did you perhaps mean 'rls` ?
```

**Comment by hrvolapeter** [comment]
I would like to take this one

**Comment by kinnison** [comment]
Hi @retep007 and thanks for being interested in helping on Rustup.

We hang out on Discord: https://discord.gg/cKdYFVz and we'd be pleased to discuss this with you on there.

**Comment by kinnison** [comment]
Closed in #1824
